### DISCUSSION TITLE
changed my password but lost it. How to get it back

### DISCUSSION THREAD

**Discussion by EtcFly** [umami-software/umami#2483] [General]
I previously deployed Umami version 1.39.5 with MySQL as the database. Now, due to server migration, I need to redeploy Umami and MySQL. I plan to import the existing MySQL data to the new MySQL server. However, I have forgotten the Umami password. How can I recover it while ensuring that the database data remains intact? Please assist me in resolving this.

**Comment by briancao** [comment]
On initialization, we default the password to `umami`.  If you have database access, you can replace your password w/ the same encrypted password to reset it to `umami`.  

That password is here: https://github.com/umami-software/umami/blob/784237bfab6da8c960efac7fd1073b8e0f67dfd9/db/postgresql/migrations/01_init/migration.sql#L199

**Reply by EtcFly** [reply]
Thanks for your help

**Reply by teledyn** [reply]
I get this error:

`ERROR 1062 (23000): Duplicate entry '41e2b680-648e-4b09-bcd7-3e2b10c06264' for key 'PRIMARY'`

and login wouldn't work, so I did "DELETE FROM user" and then the above INSERT (without quotes on user for mariadb) and I can login again! 

Something odd though: almost everywhere that asks for a new pw causes Firefox to offer a good pw, but not on this change-pw page, instead Firefox only offers other logins from sites on the same domain. Very strange.

**Reply by bskdany** [reply]
> I get this error:
> 
> `ERROR 1062 (23000): Duplicate entry '41e2b680-648e-4b09-bcd7-3e2b10c06264' for key 'PRIMARY'`
> 
> and login wouldn't work, so I did "DELETE FROM user" and then the above INSERT (without quotes on user for mariadb) and I can login again!
> 
> Something odd though: almost everywhere that asks for a new pw causes Firefox to offer a good pw, but not on this change-pw page, instead Firefox only offers other logins from sites on the same domain. Very strange.

Run this to get a new uuid
`SELECT gen_random_uuid();`

and then use that new uuid in the original query

**Comment by metasyn** [comment]
On postgres, I ran:

```SQL
UPDATE  "user" SET password='\$2b\$10\$BUli0c.muyCW1ErNJc3jL.vFRFtFJWrT8/GcR4A.sUdCznaXiqFXa' WHERE username='admin';
```

To reset it back to `umami`
